src/java.base/share/classes/java/text/MessageFormat.java

```diff
@@ -218,81 +259,99 @@
+ * "$#31,45"}. Refer to each {@code Format} subclass documentation for
+ * details.
+ *
+ * <p>Any unmatched quote is treated as closed at the end of the given
+ * pattern. For example, pattern string {@code "'{0}"} is treated as
+ * pattern {@code "'{0}'"}.
+ *
+ * <p>Any curly braces within an unquoted pattern must be balanced. For
+ * example, <code>"ab {0} de"</code> and <code>"ab '}' de"</code> are
+ * valid patterns, but <code>"ab {0'}' de"</code>, <code>"ab } de"</code>
+ * and <code>"''{''"</code> are not.
+ *
+ * <dl><dt><b>Warning:</b><dd>The rules for using quotes within message
+ * format patterns unfortunately have shown to be somewhat confusing.
+ * In particular, it isn't always obvious to localizers whether single
+ * quotes need to be doubled or not. Make sure to inform localizers about
+ * the rules, and tell them (for example, by using comments in resource
+ * bundle source files) which strings will be processed by {@code MessageFormat}.
+ * Note that localizers may need to use single quotes in translated
+ * strings where the original version doesn't have them.
+ * </dl>
+ *
+ * <h2>Usage Information</h2>
  *
  * <p>
- * Here are some examples of usage.
- * In real internationalized programs, the message format pattern and other
- * static strings will, of course, be obtained from resource bundles.
- * Other parameters will be dynamically determined at runtime.
+ * The following example demonstrates a general usage of {@code MessageFormat}.
+ * In internationalized programs, the message format pattern and other
+ * static strings will likely be obtained from resource bundles.
+ *
  * <p>
- * The first example uses the static method {@code MessageFormat.format},
- * which internally creates a {@code MessageFormat} for one-time use:
  * {@snippet lang=java :
  * int planet = 7;
  * String event = "a disturbance in the Force";
- *
  * String result = MessageFormat.format(
  *     "At {1,time} on {1,date}, there was {2} on planet {0,number,integer}.",
  *     planet, new Date(), event);
  * }
- * The output is:
- * <blockquote><pre>
- * At 12:30 PM on Jul 3, 2053, there was a disturbance in the Force on planet 7.
- * </pre></blockquote>
  *
- * <p>
- * The following example creates a {@code MessageFormat} instance that
- * can be used repeatedly:
- * {@snippet lang=java :
- * int fileCount = 1273;
- * String diskName = "MyDisk";
- * Object[] testArgs = {Long.valueOf(fileCount), diskName};
- *
- * MessageFormat form = new MessageFormat(
- *     "The disk \"{1}\" contains {0} file(s).");
- *
- * System.out.println(form.format(testArgs));
- * }
- * The output with different values for {@code fileCount}:
+ * {@code result} returns the following:
  * <blockquote><pre>
- * The disk "MyDisk" contains 0 file(s).
- * The disk "MyDisk" contains 1 file(s).
- * The disk "MyDisk" contains 1,273 file(s).
+ * At 12:30 PM on Jul 3, 2053, there was a disturbance in the Force on planet 7.
```

Comment:
An explicit date `new Date(7,3,2053)` would give predictable results between the code and the result string.

Reply:
Right, replaced with `new GregorianCalendar(2053, Calendar.JULY, 3, 12, 30).getTime()` since the Date constructor is deprecated. Also turns out the existing example output was wrong, so updated that as well.